zerver/worker/queue_processors.py

```diff
@@ -219,7 +219,10 @@ def consume(self, data):
 class ConfirmationEmailWorker(QueueProcessingWorker):
     def consume(self, data):
         # type: (Mapping[str, Any]) -> None
-        invitee = get_prereg_user_by_email(data["email"])
+        invitee = PreregistrationUser.objects.filter(id=data["prereg_id"]).first()
```

Comment:
This makes an incompatible change to the format of these queue entries, so that when upgrading across this change (e.g. from 1.7.1 to the next release), any old items in the queue will cause exceptions and be dropped.

This queue isn't likely to be big, and it isn't super high-impact if a few entries are dropped. But I think it also wouldn't be too hard to avoid -- like have a case here where if `prereg_id` is missing, we look for `email` instead, and a comment saying this is for upgrades from 1.7.1 to the next release. And that'd help set a good example for other changes to queue-entry formats.